src/test/java/com/lambdaschool/foundation/controllers/UserControllerTest.java

```diff
@@ -0,0 +1,303 @@
+import com.lambdaschool.foundation.models.User;
+import com.lambdaschool.foundation.repository.UserRepository;
+import com.lambdaschool.foundation.services.CityService;
+import com.lambdaschool.foundation.services.HelperFunctions;
+import com.lambdaschool.foundation.services.UserService;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.security.oauth2.jwt.JwtDecoder;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.RequestBuilder;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultHandlers;
+import org.springframework.web.context.WebApplicationContext;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+//@RunWith(SpringRunner.class)
+//@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT,
+//    classes = FoundationApplication.class)
+//@AutoConfigureMockMvc
+//@WithMockUser(username = "admin")
+@WebMvcTest(UserController.class)
+class UserControllerTest {
+
+    @Autowired
+    private WebApplicationContext webApplicationContext;
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @MockBean
+    private HelperFunctions helperFunctions;
+
+    @MockBean
+    private UserRepository userRepository;
+
+    @MockBean
+    private JwtDecoder jwtDecoder;
+
+    @MockBean
+    private UserService userService;
+
+    @MockBean
+    private CityService cityService;
+
+    List<User> userList;
+
+    @BeforeEach
+    void setUp() {
+        userList = new ArrayList<>();
+
+//            City c1 = new City("Washington, District of Columbia");
+//            c1.setPopulationdensityrating(20);
+//            c1.setSafteyratingscore(90);
+//            c1.setCostoflivingscore(15);
+//            c1.setAverageincome(110287d);
+//            c1.setAveragetemperature(100);
+//            c1.setLat(33.3367f);
+//            c1.setLon(-90.1234f);
+//
+//            City c2 = new City("Fort Lauderdale, Florida");
+//            c2.setPopulationdensityrating(43);
+//            c2.setSafteyratingscore(86);
+//            c2.setCostoflivingscore(12);
+//            c2.setAverageincome(39477d);
+//            c2.setAveragetemperature(70);
+//            c2.setLat(32.7673f);
+//            c2.setLon(-96.7776f);
```

Comment:
If these City objects are not needed for testing please remove. Commented out code does not belong in main branch per PR rubric.